### ISSUE TITLE
Update request: ollama 0.1.17 → 0.1.20

### ISSUE THREAD

**Issue by peperunas** [NixOS/nixpkgs#280030]
- Package name: ollama
- Latest released version: 0.1.20

- Current version on the unstable channel: 0.1.17
- Current version on the stable/release channel: 0.1.7

- [X] Checked the [nixpkgs pull requests](https://github.com/NixOS/nixpkgs/pulls)

**Notify maintainers**

@dit7ya @elohmeier 

-----

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

**Comment by trzpiot** [comment]
0.1.20 was released.[^1] :tada: Had a quick look into the source changes. Seems that patching is required because of the integration of ROCm.

[^1]: https://github.com/jmorganca/ollama/releases/tag/v0.1.20

**Comment by peperunas** [comment]
> 0.1.20 was released.[1](#user-content-fn-1-d035281a00920823637da99f333c5a87) 🎉 Had a quick look into the source changes. Seems that patching is required because of the integration of ROCm.
> ## Footnotes
> 
>     1. https://github.com/jmorganca/ollama/releases/tag/v0.1.20 [↩](#user-content-fnref-1-d035281a00920823637da99f333c5a87)

I modified the title :-)

**Comment by TanvirOnGH** [comment]
0.1.21 is going to be released very soon. https://github.com/ollama/ollama/releases/tag/v0.1.21

**Comment by peperunas** [comment]
I opened #283496

**Comment by TanvirOnGH** [comment]
update: 0.1.22 is the latest version. https://github.com/ollama/ollama/releases/tag/v0.1.22

**Comment by Casul51** [comment]
Seems update can be problematic. I've tried manually download and executing new ollama and encountered a lot of issues connected with ROCM libraries.
After adding 
```nix
hardware.opengl.extraPackages = with pkgs; [ 
  rocmPackages.clr.icd #following for GPU AI acceleration
  rocmPackages.rocm-smi
  rocmPackages.clr
  rocmPackages.hipblas
  rocmPackages.rocblas
  rocmPackages.rocsolver
  rocmPackages.rocm-comgr
  rocmPackages.rocm-runtime
  rocmPackages.rocsparse
  
# unnecessary for OLLAMA ?
  amdvlk
  rocm-opencl-icd #gaming?
  rocm-opencl-runtime #gaming?
  libva #some hardware acceleration for stuff like OBS
  vaapiVdpau
  libvdpau-va-gl
];
```
I've got it to state where it crushes:
```
ollama ➤ steam-run ./ollama-linux-amd64 serve
2024/01/31 16:33:00 images.go:857: INFO total blobs: 10
2024/01/31 16:33:00 images.go:864: INFO total unused blobs removed: 0
2024/01/31 16:33:00 routes.go:950: INFO Listening on 127.0.0.1:11434 (version 0.1.22)
2024/01/31 16:33:00 payload_common.go:106: INFO Extracting dynamic libraries...
2024/01/31 16:33:02 payload_common.go:145: INFO Dynamic LLM libraries [rocm_v5 rocm_v6 cpu_avx2 cuda_v11 cpu_avx cpu]
2024/01/31 16:33:02 gpu.go:94: INFO Detecting GPU type
2024/01/31 16:33:02 gpu.go:236: INFO Searching for GPU management library libnvidia-ml.so
2024/01/31 16:33:02 gpu.go:282: INFO Discovered GPU libraries: []
2024/01/31 16:33:02 gpu.go:236: INFO Searching for GPU management library librocm_smi64.so
2024/01/31 16:33:02 gpu.go:282: INFO Discovered GPU libraries: [/nix/store/hc6rxhy0wv0xvjaffs0ccadkjdq80h7i-rocm-smi-5.7.1/lib/librocm_smi64.so.5.0]
2024/01/31 16:33:02 gpu.go:109: INFO Radeon GPU detected
[GIN] 2024/01/31 - 16:33:07 | 200 |       29.62µs |       127.0.0.1 | HEAD     "/"
[GIN] 2024/01/31 - 16:33:07 | 200 |     475.931µs |       127.0.0.1 | POST     "/api/show"
[GIN] 2024/01/31 - 16:33:07 | 200 |     369.801µs |       127.0.0.1 | POST     "/api/show"
2024/01/31 16:33:08 cpu_common.go:11: INFO CPU has AVX2
loading library /tmp/ollama3511034227/rocm_v5/libext_server.so
2024/01/31 16:33:08 dyn_ext_server.go:90: INFO Loading Dynamic llm server: /tmp/ollama3511034227/rocm_v5/libext_server.so
2024/01/31 16:33:08 dyn_ext_server.go:145: INFO Initializing llama server
free(): invalid pointer
```
But before adding `rocmPackages.rocsparse` it seems also miss new version of `rocmPackages.hipblas` because it was asking for `libhipblas.so.2` but current unstable version have `libhipblas.so.1.1.0`

CPU works fine without mentioned ROCM packages.

**Comment by abysssol** [comment]
This appears to have been resolved by  #289108. Reopen if it's not.
